test/Libraries/RevitIntegrationTests/AsyncRunningTests.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.IO;
+using System.Linq;
+using Autodesk.Revit.DB;
+using Dynamo.Core.Threading;
+using DSCoreNodesUI.Input;
+using NUnit.Framework;
+using Revit.Elements;
+using RevitServices.Persistence;
+using RevitTestServices;
+using RTF.Framework;
+using Dynamo.Nodes;
+
+namespace RevitSystemTests
+{
+    /// <summary>
+    /// Test cases in this fixture will be supposed to run in an asyncronous mode
+    /// </summary>
+    [TestFixture]
+    public class AsyncRunningTests : RevitSystemTestBase
+    {
+        /// <summary>
+        /// Override the implementation from the base class so that the scheduler
+        /// will run the tasks asyncoronously
+        /// </summary>
+        protected override TaskProcessMode RevitTaskProcessMode
+        {
+            get
+            {
+                return TaskProcessMode.Asynchronous;
+            }
+        }
+
+        /// <summary>
+        /// Run the model but will wait for the asyncronous tasks to be completed
+        /// </summary>
+        public void AsyncRunCurrentModel()
+        {
+            RunCurrentModel();
+
+            // Run till all asyncronous tasks are completed
+            var scheduler = ViewModel.Model.Scheduler;
+            while (scheduler.ProcessNextTask(false)) { }
```

Comment:
So this is more like *process tasks when `ProcessNextTask` is called*, rather than *running tests asynchronously*, right? The `XxxAsyncTask` basically executes on the test thread instead of *asynchronously* on a separate thread. I see how this will resolve document-event interlock issue, but maybe it is not in the real sense of *asynchronous execution* :)